### ISSUE TITLE
refactor.delete does not delete nodes

### ISSUE THREAD

**Issue by antoinevastel** [jsoverson/shift-refactor#3]
Hi, thanks for open-sourcing this library. I'm currently testing it to deobfuscate different kinds of fingerprinting/security scripts. I'll try to post some of the issues I encounter during my tests, even though I can't guarantee the problem does not come from errors in my own code. 

refactor.delete seems to not always delete nodes.

```javascript
console.log(refactor.query(`VariableDeclarationStatement[declaration.declarators.length > 1][tmpAtt='todelete']`).length)
refactor.delete(`VariableDeclarationStatement[declaration.declarators.length > 1][tmpAtt='todelete']`);
console.log(refactor.query(`VariableDeclarationStatement[declaration.declarators.length > 1][tmpAtt='todelete']`).length)
```

Before and after delete is called there are 50 nodes.
The autoCleanup parameter is set to true. Nevertheless, using refactor.cleanup() does not help.

**Comment by jsoverson** [comment]
Can you include the source you are running this against?

**Comment by antoinevastel** [comment]
I shared the source in a private message on Twitter to avoid problems related to intellectual property.

**Comment by jsoverson** [comment]
Fixed in 0.3.0. The recorded insertions were not being cleaned up properly. `delete` was targeting an incorrectly inserted node in your case so it appeared as though it was doing nothing.
